clients/TypeScript/packages/client/src/ChainSync/findIntersect.ts

```diff
@@ -0,0 +1,40 @@
+import { nanoid } from 'nanoid'
+import { Ogmios, Point, Tip } from '@cardano-ogmios/schema'
+import { IntersectionNotFoundError, UnknownResultError } from '../errors'
+import { baseRequest } from '../Request'
+import { ensureSocket, InteractionContext } from '../Connection'
+
+// type Intersection = Ogmios['FindIntersectResponse']['result']['IntersectionFound']
+export type Intersection = { point: Point, tip: Tip }
+
+export const findIntersect = (points: Point[], context?: InteractionContext): Promise<Intersection> => {
+  return ensureSocket<Intersection>((socket) => {
+    return new Promise((resolve, reject) => {
+      const requestId = nanoid(5)
+      socket.once('message', (message) => {
+        const response: Ogmios['FindIntersectResponse'] = JSON.parse(message)
+        if (response.reflection.requestId !== requestId) { return }
```

Comment:
I am not sure to get the rationale of not exposing the mirror/reflection to callers for `findIntersect`, but using it internally only :thinking: 

This is also discarding messages that are not intersection responses which may accidentally lead to messages being skipped? One thing to know about `FindIntersect` is that it is not allowed to interleave it within some `RequestNext` messages; this will lead to a client error from Ogmios and isn't possible to express using the Ouroboros mini-protocols (but possible with WebSockets unfortunately).

Reply:
> I am not sure to get the rationale of not exposing the mirror/reflection to callers for findIntersect, but using it internally only

I decided to omit this capability with the local state queries too as I only see value in being able to mirror a value in `requestNext`.  Maybe I'm missing a use-case?

Reply:
A use-case I can think of is using the reflection to drive a state-machine. I used to do this in some cases with Ogmios, where I would submit a state alongside any message and then, in the request handler, switch over that state to know what handler to call as a result. Arguably, one can use the response type directly for that to distinguish FindIntersect from RequestNext requests..

Another use-case can be for example, in the state query. One may send multiple `utxo` query for different set of addresses, and put the addresses in the mirror / reflection, so that the response handler has a way to figure out to which addresses correspond the UTxOs.

Reply:
> Arguably, one can use the response type directly for that to distinguish FindIntersect from RequestNext requests.
> ... One may send multiple utxo query for different set of addresses, and put the addresses in the mirror / reflection, so that the response handler has a way to figure out to which addresses correspond the UTxOs.

The values are returned as promises, so the context is captured in a variable.

https://github.com/KtorZ/cardano-ogmios/pull/29/files#diff-92254b7a19d36bcef42fdc1c0124b4150063bcee67b9dfdea248ffcf13ad072cR217-R220

``` Typescript
it('fetches the UTxO for the given addresses', async () => {
  const utxoSet = await utxo(['addr_test1qqymtheun4y437fa6cms4jmtfex39wzz7jfwggudwnqkdnr8udjk6d89dcjadt7tw6hmz0aeue2jzdpl2vnkz8wdk4fqz3y5m9'], { connection })
  expect(utxoSet[0]).toBeDefined()
})
```

Reply:
> This is also discarding messages that are not intersection responses which may accidentally lead to messages being skipped? 

The idea here is to just ignore messages we're not expecting, as all registered handlers will receive the message. I'll do some more concurrency testing to demonstrate this.